File: src/main/java/net/hectus/neobb/turn/person_game/structure/PTStoneWall.java
package net.hectus.neobb.turn.person_game.structure;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.structure.Structure;
import net.hectus.neobb.structure.StructureManager;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.turn.default_game.structure.StructureTurn;
import net.hectus.neobb.turn.person_game.block.PTStonecutter;
import net.hectus.neobb.turn.person_game.categorization.ArmorCategory;

import java.util.List;

public class PTStoneWall extends StructureTurn implements ArmorCategory {
    public PTStoneWall(NeoPlayer player) { super(player); }
    public PTStoneWall(PlacedStructure data, NeoPlayer player) { super(data, player); }

    @Override
    public Structure referenceStructure() {
        return StructureManager.structure("stone_wall");
    }

    @Override
    public int armor() {
        return 3;
    }

    @Override
    public List<Class<? extends Turn<?>>> counteredBy() {
        return List.of(PTStonecutter.class);
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/structure/PTStoneWall.kt
package net.hectus.neobb.turn.person_game.structure

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures
import net.hectus.neobb.turn.Turn
import net.hectus.neobb.turn.default_game.structure.StructureTurn
import net.hectus.neobb.turn.person_game.block.PTStonecutter
import net.hectus.neobb.turn.person_game.categorization.ArmorCategory
import kotlin.reflect.KClass

class PTStoneWall(data: PlacedStructure?, cord: Cord?, player: NeoPlayer?) : StructureTurn(data, cord, player), ArmorCategory {
    override val staticStructure: StaticStructure = StaticStructures.Person.STONE_WALL

    override fun armor(): Int = 3

    override fun counteredBy(): List<KClass<out Turn<*>>> = listOf(PTStonecutter::class)
}
